scalafix-core/src/main/scala/scalafix/v1/SymbolAccess.scala

```diff
@@ -0,0 +1,20 @@
+package scalafix.v1
+
+import scala.meta.internal.{semanticdb => s}
+
+final class SymbolAccess private[scalafix] (a: s.Access) {
+  def isPrivate: Boolean = a.isInstanceOf[s.PrivateAccess]
+  def isPrivateThis: Boolean = a.isInstanceOf[s.PrivateThisAccess]
+  def privateWithin: Option[Symbol] = a match {
+    case s.PrivateWithinAccess(symbol) => Some(Symbol(symbol))
+    case _ => scala.None
+  }
+  def isProtected: Boolean = a.isInstanceOf[s.ProtectedAccess]
+  def isProtectedThis: Boolean = a.isInstanceOf[s.ProtectedThisAccess]
+  def protectedWithin: Option[Symbol] = a match {
+    case s.ProtectedWithinAccess(symbol) => Some(Symbol(symbol))
+    case _ => scala.None
+  }
+  def isPublic: Boolean = a.isInstanceOf[s.PublicAccess]
+  def isNone: Boolean = a == s.NoAccess
```

Comment:
An alternative approach to this API would be to replace `privateWithin` and `protectedWithin` with `isPrivateWithin`, `isProtectedWithin` and `within`. I think I like that better than the current API.

Reply:
https://github.com/scalameta/scalameta/pull/1750